serviceregistration/kubernetes/client/client.go

```diff
@@ -29,29 +29,33 @@ var (
 	ErrNotInCluster   = errors.New("unable to load in-cluster configuration, KUBERNETES_SERVICE_HOST and KUBERNETES_SERVICE_PORT must be defined")
 )
 
+// Client is a minimal Kubernetes client. We rolled our own because the existing
+// Kubernetes client-go library available externally has a high number of dependencies
+// and we thought it wasn't worth it for only two API calls. If at some point they break
+// the client into smaller modules, or if we add quite a few methods to this client, it may
+// be worthwhile to revisit that decision.
+type Client struct {
+	logger hclog.Logger
+	config *Config
+	stopCh chan struct{}
+}
+
 // New instantiates a Client. The stopCh is used for exiting retry loops
 // when closed.
-func New(logger hclog.Logger, stopCh <-chan struct{}) (*Client, error) {
+func New(logger hclog.Logger) (*Client, error) {
 	config, err := inClusterConfig()
 	if err != nil {
 		return nil, err
 	}
 	return &Client{
 		logger: logger,
 		config: config,
-		stopCh: stopCh,
+		stopCh: make(chan struct{}),
```

Comment:
Originally the `stopCh` was to break out of the retry loop if the outer environment closed down. However, since Go cleans up still-running goroutines on exit, and we don't need to save state, does it seem like we could just delete it?

If we simply trusted Go to clean up, we could probably eliminate worrying about `ctx.Done()` on line 141, and `retryableReq.WithContext(ctx)` on line 146 as well.

Reply:
We would still want to keep this because we need a way to context cancel an in-flight request if a shutdown is triggered, which can be done by the closing of this channel through `c.Shutdown()`. 

A request may or may not be done via a goroutine, but it's good to have a way for the client to cancel a request if signaled to do so. For instance, `retryHandler.Run` technically sends API calls async, but we still want a way to cancel requests if a shutdown is triggered to prevent the request from fully completing before the main process terminates.